ast/edit/task_editor.go

```diff
@@ -0,0 +1,138 @@
+package edit
+
+import (
+	"fmt"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/codes"
+)
+
+// GetOption finds and returns the AST node for the option value.
+func GetOption(pkg *ast.Package, name string) (ast.Node, error) {
+	for _, file := range pkg.Files {
+		for _, st := range file.Body {
+			val, ok := st.(*ast.OptionStatement)
+			// wrong statement type; skip this statement
+			if !ok {
+				continue
+			}
+			assign := val.Assignment
+			va, ok := assign.(*ast.VariableAssignment)
+			if va.ID.Name == name {
+				if ok {
+					return val, nil
+				}
+			}
+		}
+	}
+	return nil, &flux.Error{
+		Code: codes.Internal,
+		Msg:  "Option not found",
+	}
+}
+
+// SetOption replaces an existing option definition with the provided option node or adds
+// the option if it doesn't exist. The pkg AST is mutated in place.
+func SetOption(pkg *ast.Package, name string, id string, opt ast.Node) error {
```

Comment:
Should we change the API so that is deals with `ast.File` instead of `ast.Package` that way there is no ambiguity around which file to modify? 

In other words leave the logic of finding the specific file outside of this API.

Reply:
Yep, that's a good idea.

Reply:
I'll need to rewrite some of my tests to make this work, but this change makes sense.